explore/MultiTaskClassification.py

```diff
@@ -0,0 +1,461 @@
+
+genes_Cabozantinib = {
+    '2322':'FLT3',
+    '3815':'KIT',
+    '4233':'MET',
+    '5979':'RET',
+    '3791':'KDR', #VEGFR2 
+    }
+
+# Crizotinib (Xalkori)
+
+genes_Crizotinib = {
+    '238':'ALK',
+    '4233':'MET',    
+    '6098':'ROS1',
+    }
+
+# Imatinib (Gleevec)
+
+genes_Imatinib = {
+    '3815':'KIT',
+    '5159':'PDGFRB',
+    '25':'ABL',
+    }
+
+# Palbociclib (Ibrance)
+
+genes_Palbociclib = {
+    '1019':'CDK4',
+    '1021':'CDK6',
+    }
+
+# Pazopanib (Votrient)
+
+genes_Pazopanib = {
+    '3791':'KDR',
+    '5159':'PDGFRB',
+    '3815':'KIT',
+    }
+
+# Ponatinib (Iclusig)
+
+genes_Ponatinib = {
+    '25':'ABL',
+    '2263':'FGFR1-3',
+    '2322':'FLT3',
+    '3791':'KDR',
+    }
+
+# Regorafenib (Stivarga)
+
+genes_Regorafenib = {
+    '3815':'KIT',
+    '5159':'PDGFRB',
+    '5894':'RAF1',    
+    '5979':'RET',
+    '2321':'FLT1', #VEGFR1
+    '3791':'KDR', #VEGFR2    
+    '2324':'FLT4', #VEGFR3
+    }
+
+# Sorafenib (Nexavar)
+
+genes_Sorafenib = {
+    '3791':'KDR',
+    '5159':'PDGFRB',    
+    '3815':'KIT',
+    '5894':'RAF1', 
+    }
+
+# Vandetanib (Caprelsa)
+
+genes_Vandetanib = {
+    '1956':'EGFR',
+    '5979':'RET',
+    '3791':'KDR',
+    }
+
+# Vismodegib (Erivedge)
+
+genes_Vismodegib = {
+    '5727':'PTCH1',
+    '6608':'SMO',
+    }
+
+# Ziv-aflibercept (Zaltrap)
+
+genes_Zaltrap = {
+    '5281':'PIGF',
+    '7422':'VEGFA',
+    '7423':'VEGFB',
+    }
+
+
+# In[8]:
+
+# all targted genes, needed for comparison, store them in y
+
+genes = {
+    '1956':'EGFR',    
+    '2064':'HER2',
+    '3815':'KIT', 
+    '5159':'PDGFRB',
+    '2321':'FLT1', #VEGFR1
+    '3791':'KDR', #VEGFR2    
+    '2324':'FLT4', #VEGFR3
+    '2322':'FLT3',
+    '4233':'MET',
+    '5979':'RET',
+    '238':'ALK',    
+    '6098':'ROS1',
+    '25':'ABL',
+    '1019':'CDK4',
+    '1021':'CDK6',
+    '2263':'FGFR1-3',
+    '5894':'RAF1',    
+    '5727':'PTCH1',
+    '6608':'SMO',
+    '5281':'PIGF',
+    '7422':'VEGFA',
+    '7423':'VEGFB',
+    } 
+
+y=Y[list(genes.keys())] 
+
+
+# In[9]:
+
+# Using same training set for all targets
+# Set aside 10% of the data for testing, this can only be done where the number of mutations is large enough
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
+'Size: {:,} features, {:,} training samples, {:,} testing samples'.format(len(X.columns), len(X_train), len(X_test))
+
+features_train=pipeline.fit_transform(X=X_train, y=y_train) 
+features_test=pipeline.fit_transform(X=X_test, y=y_test)
+
+
+# In[10]:
+
+# Results for all genes when observed separately, accuracy score is used for evaluation
+
+as_RF =[]
+as_L = []
+ps_RF =[]
+ps_L = []
+
+for GENE in list(genes.keys()):
+    
+    y_train_each = y_train[GENE]
+    true_label = list(y_test[GENE])
+    
+    y_test_predict = dict()
+    for name, classifier in Classifiers_mono.items():
+        classifier.fit(features_train, y_train_each)
+        y_test_predict[name] = classifier.predict(features_test)

```

Comment:
Thank you guys so much for your comments. I am really sorry for this huge delay; finishing up my dissertation is not leaving me enough time to do other things. 
Regarding new files, I corrected the fit-transform issue and I added AUROC values and CV(cross validation) form of classifiers. Also, I made csv file with drug and gene information so someone can use it for future research. 
Unfortunately, I am not happy with the ROC curves and I am not sure how to improve them. If you think it is not appropriate to leave this analysis in this condition, please let me know and I will delete these files.